parquet-thrift/src/main/java/org/apache/parquet/thrift/ThriftSchemaConverter.java

```diff
@@ -57,7 +57,13 @@ public MessageType convert(Class<? extends TBase<?, ?>> thriftClass) {
   }
 
   public MessageType convert(StructType struct) {
-    MessageType messageType = ThriftSchemaConvertVisitor.convert(struct, fieldProjectionFilter);
+    MessageType messageType = ThriftSchemaConvertVisitor.convert(struct, fieldProjectionFilter, true);
+    fieldProjectionFilter.assertNoUnmatchedPatterns();
+    return messageType;
+  }
+
+  public MessageType convertWithoutProjection(StructType struct) {
```

Comment:
Maybe add a comment to state the struct being passed in should be file schema stored in json, and the isUnion method will nor be called...
Basically just document why it's separated from the normal convert code path in case we forget it in the future